src/dvc_data/hashfile/build.py

```diff
@@ -111,59 +266,61 @@ def _build_tree(
 
     path = path.rstrip(fs.sep)
 
-    if ignore:
-        walk_iter = ignore.walk(fs, path)
-    else:
-        walk_iter = fs.walk(path)
-
-    tree_meta = Meta(size=0, nfiles=0, isdir=True)
-    # assuring mypy that they are not None but integer
-    assert tree_meta.size is not None
-    assert tree_meta.nfiles is not None
-
     tree = Tree()
+    size = 0
+    for root, files in _walk_files(fs, path, ignore=ignore):
```

Comment:
Also changed this to `walk` with `detail=True` which will avoid the need for `fs.info()` calls. On local minio, `_build_tree()` drops from ~60s to ~7s.

Reply:
The downside is that we have to now work with file_infos (a dict of `{filename: info}`), rather than filename itself.